pkg/docker/docker.go

```diff
@@ -900,8 +894,21 @@ func (d *stiDocker) holdHijackedConnection(tty bool, inputStream io.Reader, outp
 }
 
 // RunContainer creates and starts a container using the image specified in opts
-// with the ability to stream input and/or output.
+// with the ability to stream input and/or output.  Any non-nil
+// opts.Std{in,out,err} will be closed upon return.
 func (d *stiDocker) RunContainer(opts RunContainerOptions) error {
+	// Ensure that Std{in,out,err} are closed upon return, including under error
+	// circumstances.
```

Comment:
Once RunContainer returns, by definition nothing more can be written to std{out,err}, therefore it closes them.  It also closes stdin so that any goroutine still trying to write to the container will error out.

sti.go:685 uses a goroutine + (unbuffered) pipe + *string (should actually be a bytes.Buffer) to keep the stderr data for reading, so we should be good.